pandas/core/nanops.py

```diff
@@ -864,6 +866,12 @@ def reduction(
                 result.fill(np.nan)
             except (AttributeError, TypeError, ValueError):
                 result = np.nan
+        elif is_object_dtype(dtype) and values.ndim == 1 and na_mask.any():
+            # Need to explicitly mask NA values for object dtypes
+            if skipna:
+                result = getattr(values[~na_mask], meth)(axis)
```

Comment:
This masking could also be done in the `min`/`max` functions? (as you had before?)

Or, another option might be to add a min/max function to `mask_ops.py`, similarly as I am doing for `sum` in https://github.com/pandas-dev/pandas/pull/30982 (but it should be simpler for min/max, as those don't need to handle the `min_count`)

Reply:
I think a benefit of having it here is that this also fixes a bug for Series: `pd.Series(["a", np.nan]).min()` currently raises even though it shouldn't

Reply:
Ah, that's a good point. Can you add a test for that, then? 

Now, that aside, I think longer term we still want the separate min/max in mask_ops.py, so it can also be used for the int dtypes. But that can then certainly be done for a separate PR.